leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```diff
@@ -0,0 +1,260 @@
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Orange - Add better support for Arguments of Execute Operation.
+ *******************************************************************************/
+package org.eclipse.leshan.core.request.execute;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.leshan.core.util.StringUtils;
+
+/**
+ * Arguments for Execute Operation.
+ */
+public class Arguments implements Iterable<Argument> {
+
+    private final List<Argument> argumentList;
+
+    /**
+     * Arguments should be build using {@link ArgumentsBuilder}.
+     */
+    private Arguments(List<Argument> argumentList) {
+        this.argumentList = argumentList;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Arguments [argumentList=%s]", argumentList.toString());
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((argumentList == null) ? 0 : argumentList.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Arguments other = (Arguments) obj;
+        if (argumentList == null) {
+            if (other.argumentList != null)
+                return false;
+        } else if (!argumentList.equals(other.argumentList))
+            return false;
+        return true;
+    }
+
+    /**
+     * Returns number of arguments.
+     */
+    public int size() {
+        return argumentList.size();
+    }
+
+    /**
+     * Check if Arguments has any elements.
+     */
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    /**
+     * Parse String into {@link Arguments}
+     *
+     * <pre>
+     * {@code
+     * arglist = arg *( "," arg )
+     * arg = DIGIT / DIGIT "=" "'" *CHAR "'"
+     * DIGIT = "0" / "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9"
+     * CHAR = "!" / %x23-26 / %x28-5B / %x5D-7E
+     * }
+     * </pre>
+     * 
+     * @param arglist String contains arguments to parse.
+     * @return {@link Arguments} represents list of arguments.
+     * @throws InvalidArgumentException if text has invalid format.
+     */
+    static Arguments parse(String arglist) throws InvalidArgumentException {
+        List<Argument> argumentList = new ArrayList<>();
+
+        if (arglist != null && !arglist.isEmpty()) {
+            int beginProbe = 0;
+            int validBegin = 0;
+            while (true) {
```

Comment:
I'm not sure what is your opinion about this but generally I try to avoid `while(true)` loop ?

Reply:
For me splitting logic into variable assigment (separatorIndex) and loop breaking looks simpler than put the two things together. And variable is used only inside loop so there is no need to declare one outside loop.

For me I don't see problem with ```while(true)``` but I can change this if you want.

Reply:
Not a big deal, let's go like this. :slightly_smiling_face: